### DISCUSSION TITLE
Query vm by custom attribute

### DISCUSSION THREAD

**Discussion by lelvisl** [vmware/govmomi#2920] [Community Help] [answered]
Hi!
How i can query vm from vCenter by some custom attribute?

**Comment by embano1** [comment, marked as answer]
IMHO this is not possible and you need to retrieve the `ManagedObjectReferences` first and then perform attribute matching, like `govc` does https://github.com/vmware/govmomi/blob/master/govc/fields/info.go#L88

**Reply by lelvisl** [reply]
I need to query 10 vms from vc with more than thousands vms, it's not so optimal. 
There is no server-side filtering?
